R/assertions.R

```diff
@@ -62,16 +107,14 @@ get_assert_method <- function(predicate, method = list(direct = assertr::assert,
 #' @export
 #' @seealso validate_cols validate_rows
 validate_if <- function(data, expr, description = NA, obligatory = FALSE, skip_chain_opts = FALSE,
-                      success_fun = assertr::success_append,
-                      error_fun = assertr::error_append, defect_fun = assertr::defect_append) {
-  assertr::verify(data = data,
-                  expr = !!rlang::enexpr(expr),
-                  description = description,
-                  skip_chain_opts = skip_chain_opts,
-                  obligatory = obligatory,
-                  success_fun = success_fun,
-                  error_fun = error_fun,
-                  defect_fun = defect_fun)
+                        success_fun = assertr::success_append,
+                        error_fun = assertr::error_append, defect_fun = assertr::defect_append) {
+
+  assertr::verify(data = data, expr = !!rlang::enexpr(expr),
+                  description = description, skip_chain_opts = skip_chain_opts,
+                  obligatory = obligatory, success_fun = success_fun,
+                  error_fun = error_fun, defect_fun = defect_fun) %>%
+    check_assertr_expression(data, description, error_fun)
```

Comment:
I wanted that `check_assertr_expression` takes the `data`, `description`, and `error_fun` from the parent environment, to not move those around all the time, but it was kinda not working for me. It might be that being inside `tryCatch` error handling was too deep. Any ideas?

Reply:
I'm ok with passing those as arguments. It makes the code easier to understand (compared to the "magic" of picking them from the parent environment).